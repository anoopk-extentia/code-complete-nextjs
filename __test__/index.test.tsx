/**
 * Filename: index.test.tsx \
 * Description: This is the index page test case file.\
 * Extentia: Copyright (c) 2022
 */

import { createRenderer } from 'react-test-renderer/shallow';
import { cleanup } from '@testing-library/react';
import Home from 'pages';

const renderer = createRenderer();

afterEach(cleanup);

describe('Home Page', () => {
    it('should render home page', () => {
        renderer.render(<Home />);
        const renderedOutput = renderer.getRenderOutput();
        expect(renderedOutput).toMatchSnapshot();
    });
});
