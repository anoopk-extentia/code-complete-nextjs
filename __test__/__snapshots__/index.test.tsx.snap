// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render home page 1`] = `
<React.Fragment>
  <Head>
    <title>
      Create Next App
    </title>
    <meta
      content="Generated by create next app"
      name="description"
    />
    <link
      href="/favicon.ico"
      rel="icon"
    />
  </Head>
  <main
    className="mx-[40px] mt-[64px]"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Modal
    </button>
    <ModalComponent
      handleShowModal={[Function]}
      showModal={false}
    />
  </main>
</React.Fragment>
`;
